src/CoinLockerSystem.java

```diff
@@ -134,24 +103,125 @@ private Locker getAvailableLocker(LockerSize size) {
                     largeLockers.remove(locker);
+                    break;
                 }
-                break;
         }
-        if(locker != null){
+        if (locker != null) {
             usingLockers.put(locker.getId(), locker);
         }
         return locker;
     }
+
+    private void getMoney(int price) {
+        double paid = 0;
+        do {
+            System.out.println("Insert " + (price - paid) + " dollars.");
+            paid += scanner.nextDouble();
+            scanner.nextLine();
+        } while (paid < price);
+        if (paid > price) {
+            System.out.println("Here is your change " + (paid - price) + " dollars");
+        }
+    }
+
+    private void savePassword(Locker locker) {
+        System.out.println("Enter a password for your locker");
+        String password = scanner.nextLine();
+        locker.setPassword(password);
+    }
+
+    private void openLocker() {
+        Locker locker = null;
+        System.out.println("Enter your locker number.");
+        int lockerNum = scanner.nextInt();
+        scanner.nextLine();
+        try {
+            locker = openLocker(lockerNum);
+        } catch (IllegalArgumentException e) {
+            System.out.println("Invalid locker number.");
+            System.out.println("1. Enter the number again");
+            System.out.println("2. Cancel");
+            int choice = scanner.nextInt();
+            if (choice == 1) {
+                openLocker();
+            } else {
+                return;
+            }
+        }
+        try {
+            checkPassword(locker);
+        } catch (InvalidParameterException e) {
+            System.out.println("You entered wrong password 3 times. Please contact to the office.");
+            return;
+        }
+        System.out.println("Locker number " + lockerNum + " is opened!");
+        returnLocker(locker);
+    }
+
+    Locker openLocker(int lockerNum) {
+        if (!usingLockers.containsKey(lockerNum)) {
+            throw new IllegalArgumentException(
+                    String.format("Given locker number [%d] doesn't exist.", lockerNum));
+        }
+        return usingLockers.get(lockerNum);
+    }
+
+    private void checkPassword(Locker locker) {
+        System.out.println("Enter the password for this locker");
+        int count = 0;
+        while (count < 3) {
+            String password = scanner.nextLine();
+            if (password.equals(locker.getPassword())) {
+                return;
+            }
+            System.out.println("Password is not matched. Try again.");
+            count++;
+        }
+        if (count == 3) {
+            throw new InvalidParameterException("Entered password is not matched over 3 times.");
+        }
+    }
+
+    private void returnLocker(Locker locker) {
+        usingLockers.remove(locker.getId());
+        String lockerType = locker.getClass().getName();
+        if (lockerType.equals("SmallLocker")) {
```

Comment:
just fyi, string could also be used in switch-case

Reply:
Yeah-ah-p